hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/bootstrap/PartitionUtils.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client.bootstrap;
+
+import org.apache.hudi.AvroConversionUtils;
+import org.apache.hudi.HoodieSparkUtils;
+import org.apache.hudi.SparkAdapterSupport$;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.hadoop.CachingPath;
+
+import org.apache.avro.Schema;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.spark.sql.execution.datasources.SparkParsePartitionUtil;
+import org.apache.spark.sql.internal.SQLConf;
+
+public class PartitionUtils {
+
+  public static Object[] getPartitionFieldVals(Option<String[]> partitionFields,
+                                               String partitionPath,
+                                               String bootstrapBasePath,
+                                               Schema writerSchema,
+                                               Configuration hadoopConf) {
+    if (!partitionFields.isPresent()) {
+      return new Object[0];
+    }
+    SparkParsePartitionUtil sparkParsePartitionUtil = SparkAdapterSupport$.MODULE$.sparkAdapter().getSparkParsePartitionUtil();
+    return HoodieSparkUtils.parsePartitionColumnValues(
+        partitionFields.get(),
+        partitionPath,
+        new CachingPath(bootstrapBasePath),
```

Comment:
I got rid of the orElse. Since it is only called in the if block that `upsertHandle.baseFileForMerge().getBootstrapBaseFile().isPresent()`, it means that there will always be a bootstrap base path.